File: VB/ChangeEditMask.Module.Web/Controllers/WebChangeMaskController.vb
Imports ChangeEditMask.Module.Controllers
Imports DevExpress.ExpressApp.Web.Editors.ASPx
Imports DevExpress.Web

Namespace ChangeEditMask.Module.Web.Controllers

    Public Class WebChangeMaskController
        Inherits ChangeMaskControllerBase

        Protected Overrides Sub SetControlMaskSettings(ByVal propertyEditor As DevExpress.ExpressApp.Editors.PropertyEditor, ByVal mask As BusinessObjects.EditMask)
            If TypeOf propertyEditor Is ASPxStringPropertyEditor Then
                Dim textEdit As ASPxTextBox = TryCast(CType(propertyEditor, ASPxStringPropertyEditor).Editor, ASPxTextBox)
                If textEdit IsNot Nothing Then
                    Select Case mask
                        Case BusinessObjects.EditMask.Date
                            textEdit.MaskSettings.Mask = "MM/dd/yyyy"
                        Case BusinessObjects.EditMask.Time
                            textEdit.MaskSettings.Mask = "hh:mm tt"
                        Case BusinessObjects.EditMask.Numeric
                            textEdit.MaskSettings.Mask = "0999999999"
                        Case BusinessObjects.EditMask.String
                            textEdit.MaskSettings.Mask = ""
                    End Select
                End If
            End If
        End Sub
    End Class
End Namespace

File: CS/XPO/DynamicMask/DynamicMask.Blazor.Server/Controllers/WebChangeMaskController.cs
using DevExpress.ExpressApp.Blazor.Editors;
using DevExpress.ExpressApp.Blazor.Editors.Adapters;
using DynamicMask.Module.BusinessObjects;
using DynamicMask.Module.Controllers;

namespace ChangeEditMask.Module.Web.Controllers {
    public class BlazorChangeMaskController : ChangeMaskControllerBase {
        protected override void SetControlMaskSettings(DevExpress.ExpressApp.Editors.PropertyEditor propertyEditor, EditMask mask) {
            if (propertyEditor is StringPropertyEditor stringEditor) {

                //    ASPxTextBox textEdit = ((ASPxStringPropertyEditor)propertyEditor).Editor as ASPxTextBox;
                //var t = stringEditor.Control as DxTextBoxAdapter;
                //var t2 = t.ComponentModel;
                //    if (textEdit != null) {
                switch (mask) {
                    case EditMask.Date:
                        stringEditor.EditMask = "MM/dd/yyyy";
                        break;
                    case EditMask.Time:
                        stringEditor.EditMask = "hh:mm tt";
                        break;
                    case EditMask.Numeric:
                        //   stringEditor.EditMask = "(000) 000-0000";
                        stringEditor.Model.EditMask = "(000) 000-0000";
                        View.Refresh();
                        break;
                    case EditMask.String:
                        break;
                }
                //    }
            }
        }
    }
}
